### PR TITLE
Excelsior pr

### PR DESCRIPTION
**Pull request by wangxdflight**
Resubmit changes to enable px4 on excelsior after rebasing

### CODE REVIEW

File: src/modules/mavlink/mavlink_orb_subscription.cpp [hunk 1]
```diff
@@ -170,8 +170,7 @@ MavlinkOrbSubscription::is_published()
 	// We are checking now
 	_last_pub_check = now;
 
-#if defined(__PX4_QURT) || defined(__PX4_POSIX_EAGLE)
-
+#if defined(__PX4_QURT) || defined(__PX4_POSIX_EAGLE) || defined(__PX4_POSIX_EXCELSIOR)
 	// Snapdragon has currently no support for orb_exists, therefore
 	// we're not using it.
```
Comment (mcharleb) [inline src/modules/mavlink/mavlink_orb_subscription.cpp]: @wangxdflight orb_exists() is now supported upstream

**Comment by LorenzMeier** [comment]
The ifdef magic here is horrible - but unrelated to this PR. Let me try to take a pass at fixing it up. It was meant as a quick fix but got out of hand.

**Comment by LorenzMeier** [comment]
@mcharleb Will this fix our locking issues?

**Comment by wangxdflight** [comment]
I think so. I saw the same error in a different scenario , but not any more
with this change.

**Comment by LorenzMeier** [comment]
@Stifael Please test Monday on-hardware and verify that this fixes your param locking issue.

**Comment by dagar** [comment]
Should we add this to the builds?

**Comment by LorenzMeier** [comment]
@dagar Would be good.

**Comment by LorenzMeier** [comment]
@Stifael Can you test?

**Comment by LorenzMeier** [comment]
@ChristophTobler FYI

**Comment by ChristophTobler** [comment]
This fix works for me. I can now set the parameters both ways

**Comment by Stifael** [comment]
Yes I confirm that this fixes the shemlock error.

**Pull request closed without merging**